### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by paradise**

### CODE REVIEW

File: settings.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+HistoryService:
+    Db: 
+        PostgresDataConnString:
+            settings-key: HistoryService-PostgresDataConnString
```
Comment (samodovdi) [inline settings.yaml:4]: add types?

File: src/Lykke.Service.History/Settings/CqrsSettings.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Lykke.Service.History.Settings
+{
+    public class CqrsSettings
+    {
+        public string RabbitConnString { get; set; }
```
Comment (polonzer) [inline src/Lykke.Service.History/Settings/CqrsSettings.cs]: [AmqpCheck]

File: src/Lykke.Service.History/Controllers/HistoryController.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{
+    [Route("api/history")]
+    [ApiController]
+    public class HistoryController : ControllerBase
+    {
+        private readonly IHistoryRecordsRepository _historyRecordsRepository;
+
+        public HistoryController(IHistoryRecordsRepository historyRecordsRepository)
+        {
+            _historyRecordsRepository = historyRecordsRepository;
```
Comment (polonzer) [inline src/Lykke.Service.History/Controllers/HistoryController.cs:18]: I would add check for not-null here.

File: src/Lykke.Service.History/Controllers/OrdersController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+{
+    [Route("api/orders")]
+    [ApiController]
+    public class OrdersController : ControllerBase
+    {
+        private readonly IOrdersRepository _ordersRepository;
+
+        public OrdersController(IOrdersRepository ordersRepository)
+        {
+            _ordersRepository = ordersRepository;
```
Comment (polonzer) [inline src/Lykke.Service.History/Controllers/OrdersController.cs:19]: I would add check for not-null here.

File: src/Lykke.Service.History/Workflow/Handlers/EthereumCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                _logger.Warning($"Transaction hash was not set, ETH cashout", context: new { id = id, hash = command.TransactionHash });
+            }
+
+            return CommandHandlingResult.Ok();
+        }
+    }
+
+    [MessagePackObject(keyAsPropertyName: true)]
+    public class SaveEthInHistoryCommand
```
Comment (polonzer) [inline src/Lykke.Service.History/Workflow/Handlers/EthereumCommandHandler.cs]: Please move this class (as well as the other bellow) to the separate file.

File: src/Lykke.Service.History/Workflow/Projections/OrderEventProjection.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            _logger = logFactory.CreateLog(this);
+        }
+
+        public async Task<CommandHandlingResult> Handle(OrderPlacedEvent @event)
+        {
+            var entity = Mapper.Map<OrderEvent>(@event);
+
+            if (!await _historyRecordsRepository.TryInsertAsync(entity))
+            {
+                _logger.Warning($"Skipped duplicated cashin record", context: new { id = entity.Id, orderId = @event.OrderId, type = "placed" });
```
Comment (polonzer) [inline src/Lykke.Service.History/Workflow/Projections/OrderEventProjection.cs]: >duplicated cashin 
'order placed'?

File: src/Lykke.Service.History/Workflow/Projections/OrderEventProjection.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            return CommandHandlingResult.Ok();
+        }
+
+        public async Task<CommandHandlingResult> Handle(OrderCancelledEvent @event)
+        {
+            var entity = Mapper.Map<OrderEvent>(@event);
+
+            if (!await _historyRecordsRepository.TryInsertAsync(entity))
+            {
+                _logger.Warning($"Skipped duplicated cashin record", context: new { id = entity.Id, orderId = @event.OrderId, type = "cancelled" });
```
Comment (polonzer) [inline src/Lykke.Service.History/Workflow/Projections/OrderEventProjection.cs]: >duplicated cashin 
'order cancelled'?

File: client/Lykke.Service.History.Client/IHistoryApi.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using JetBrains.Annotations;
+using Refit;
+
+namespace Lykke.Service.History.Client
+{
+    /// <summary>
+    /// History client API interface.
+    /// </summary>
+    [PublicAPI]
+    public interface IHistoryApi
```
Comment (polonzer) [inline client/Lykke.Service.History.Client/IHistoryApi.cs]: I guess this interface should be extended (it is empty now).

**Review by polonzer — COMMENTED** [review]
Please remove unused directives in files (using resharper).

File: settings.yaml (hunk 1 above)
Comment (paradise) [reply settings.yaml:4]: afaik, we do not have types for PostgresSQL

**Review by polonzer — APPROVED** [review]

**Pull request merged**